test/new-e2e/tests/process/testing.go

```diff
@@ -136,6 +185,17 @@ func findProcess(
 	return found, populated
 }
 
+// filterProcesses returns processes which match the given process name
```

Comment:
I broke this out from `findProcess()` as to allow for better logging of when we don't find data. This way we don't need to log all payloads if we are missing just CPU/Memory data from the process when it's missing.